tests/shared/src/test/scala/epollcat/TcpSuite.scala

```diff
@@ -97,12 +97,12 @@ class TcpSuite extends EpollcatSuite {
           _ <- IO(assertEquals(res, "pong"))
         } yield ()
 
-        serverCh.bind(new InetSocketAddress(0)) *> server.both(client).void
+        serverCh.bind(new InetSocketAddress("localhost", 0)) *> server.both(client).void
```

Comment:
Heh, it's a good strategy. I apply it ~~mostly~~ sometimes, usually when patching small bugs. Today I cheated!

Reply:
Btw, the initial macOS failures were when using `new InetSocketAddress(0)` by itself. Each of the following work:
- `new InetSocketAddress("localhost", 0)`
- `new InetSocketAddress("127.0.0.1", 0)`
- `new InetSocketAddress("0.0.0.0", 0)`
- `new InetSocketAddress("0:0:0:0:0:0:0:0")`

So there is something strange going on in the no-argument case. After investigating aimlessly, I determined it wasn't important enough to block at this stage. And is possibly an issue in Scala Native core, and not here anyway.

Reply:
The no argument case means that the underlying code is supposed to 
use the "wildcard" address.  On SN 0.4.n, this would be the IPv4
0.0.0.0.  I think that this is an SN 0.4.0 issue.

I would have check why the one arg case and the 3rd case in your
list 0.0.0.0 differ. I would have expected them to be the same.

What errors were you seeing.

Reply:
The error I observed was that the client socket was unable to connect to the server: `ConnectionException`. But only on macOS.